pysidt/sidt.py

```diff
@@ -932,6 +946,529 @@ def regularize(self, data=None, check_data=True):
             self.r_morph,
         )
 
+class MultiEvalSubgraphIsomorphicDecisionTreeBinaryClassifier(MultiEvalSubgraphIsomorphicDecisionTree):
```

Comment:
I think a much better and standard way to organize these class dependency is to 
1. move all the shared function into a base parent class `MultiEvalSubgraphIsomorphicDecisionTree`. Use the `@abstractmethod` on the functions that are required to be defined in the child class.
2. rename the original `MultiEvalSubgraphIsomorphicDecisionTree` class to `MultiEvalSubgraphIsomorphicDecisionTreeRegressor`
3. Inherent from the new `MultiEvalSubgraphIsomorphicDecisionTree` class

To my opinion it's cleaner and easier to understand that way, and more analogous to how sklearn organizer their code.